doc/sphinx/proof-engine/tactics.rst

```diff
@@ -898,206 +1111,154 @@ Applying theorems
-   number of other non-dependent premises in the type of :token:`term` and of
-   the non-dependent premises of the type of :token:`ident`. If the conclusion
-   of the type of :token:`term` does not match the goal *and* the conclusion
-   is an inductive type isomorphic to a tuple type, then
-   the tuple is (recursively) decomposed and the first component of the tuple
-   of which a non-dependent premise matches the conclusion of the type of
-   :token:`ident`. Tuples are decomposed in a width-first left-to-right order
-   (for instance if the type of :g:`H1` is :g:`A <-> B` and the type of
-   :g:`H2` is :g:`A` then :g:`apply H1 in H2` transforms the type of :g:`H2`
-   into :g:`B`). The tactic :tacn:`apply` relies on first-order pattern matching
-   with dependent types.
-
-   .. exn:: Statement without assumptions.
-
-      This happens if the type of :token:`term` has no non-dependent premise.
-
-   .. exn:: Unable to apply.
-
-      This happens if the conclusion of :token:`ident` does not match any of
-      the non-dependent premises of the type of :token:`term`.
-
-   .. tacv:: apply {+, @term} in {+, @ident}
-
-      This applies each :token:`term` in sequence in each hypothesis :token:`ident`.
+.. _managingthelocalcontext:
 
-   .. tacv:: apply {+, @term with @bindings} in {+, @ident}
+Managing the local context
+------------------------------
 
-      This does the same but uses the bindings to instantiate
-      parameters of :token:`term` (see :ref:`bindings`).
+.. tacn:: intro {? @ident } {? @where }
 
-   .. tacv:: eapply {+, @term {? with @bindings } } in {+, @ident}
+   Applies the :tacn:`hnf` tactic until it finds an item that can be
+   introduced in the context by removing certain constructs in the goal.
+   If no item is found, the tactic fails.  The name used is
+   :n:`@ident` (if specified) or from the construct, except that if the name from the
+   construct already exists in the :term:`local context`, Coq uses a fresh name
+   instead.  The constructs have these forms:
+   (See examples :ref:`here <intro_examples>`.)
 
-      This works as :tacn:`apply … in` but turns unresolved bindings into
-      existential variables, if any, instead of failing.
+   :n:`forall x : T, @term`
+     If `x` is used in :n:`@term`, `x : T` is a :term:`dependent premise`.  Removes
+     `forall x : T,` from the goal and adds `x : T` to the context.
 
-   .. tacv:: apply {+, @term {? with @bindings } } in {+, @ident {? as @simple_intropattern}}
-      :name: apply … in … as
+   :n:`A -> …`
+     `A` is a :term:`non-dependent premise`.  Removes `A ->` from
+     the goal and adds `H : A` to the context.
 
-      This works as :tacn:`apply … in` but applying an associated
-      :token:`simple_intropattern` to each hypothesis :token:`ident`
-      that comes with such clause.
+   :n:`let x := c, @term`
+     If `x` is used in :n:`@term`, removes `let x := c,` from the goal and adds `x := c : T`
```

Comment:
I discover that you have added "If x is used in term" in your last commit. It happens that in the case of the `intro` tactic, it doesn't matter whether it is or is not used, it will be introduced in the context in all cases. (This matters for `intros *` vs `intros **`.)

Reply:
I must have tried it with `intros *`.

Reply:
@jfehrle It looks like you haven't fixed this (it still says "If `x` is used").

Reply:
Got it.  I fixed the one on the "forall" line (which I added in the "last commit", should have done both.  Updated.